framework/web/Request.php

```diff
@@ -177,6 +182,12 @@ class Request extends \yii\base\Request implements RequestInterface
      * @see getBodyParams()
      */
     public $parsers = [];
+    /**
+     * @var string name of the class to be used for uploaded file instantiation.
+     * This class should implement [[UploadedFileInterface]].
+     * @since 2.1.0
+     */
+    public $uploadedFileClass = UploadedFile::class;
```

Comment:
Potentially such cases could be handled via container i.e. `$di->get(UploadedFileInterface::class)` but I think it's good as is for 2.1.